arrebol/worker/auth/auth.go

```diff
@@ -2,68 +2,91 @@ package auth
 
 import (
 	"encoding/json"
-	"errors"
+	"github.com/google/logger"
 	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker"
 	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/auth/allowlist"
 	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/auth/token"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/auth/tolerant"
 	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/key"
 	"github.com/ufcg-lsd/arrebol-pb/crypto"
+	"os"
+	"strconv"
 )
 
-type Authenticator struct {
-	allowlist allowlist.AllowList
+const AllowListConfKey = "ALLOW_ALL"
+
+type Authorizer interface {
+	Authorize(token *token.Token) error
+}
+
+type Authenticator interface {
+	Authenticate(rawPublicKey string, signature []byte, worker *worker.Worker) (token.Token, error)
 }
 
-func NewAuth() *Authenticator {
-	allowlist := allowlist.NewAllowList()
-	var auth = Authenticator{
-		allowlist: allowlist,
+type Auth struct {
+	Authorizer Authorizer
+	Authenticator Authenticator
+}
+
+func NewAuth() *Auth {
+	return &Auth{
+		Authorizer:    NewAuthorizer(),
+		Authenticator: NewAuthenticator(),
+	}
+}
+
+func NewAuthorizer() Authorizer {
+	allow, err := strconv.ParseBool(os.Getenv(AllowListConfKey))
+
+	if err != nil {
+		logger.Fatalf("Cannot understand the flag: %s", err.Error())
+	}
+	if allow {
+		return allowlist.NewAuthorizer()
```

Comment:
I don't like the authorizer inside the allowlist file. The authorizer deals with token, for example,. We should or extract the token thing to be a common part of all authorizers, or take the authorizer out of the allowlist file, what do you thing?

Reply:
check the new design.